test/integration/operator_policy_complianceconfig_test.go

```diff
@@ -0,0 +1,208 @@
+
+			By("Create namespace " + subscriptionNamespace1)
+			_, err = common.OcManaged(
+				"create", "namespace", subscriptionNamespace1,
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create policy " + policyName + " which contains " + operatorPolicyName)
+			common.DoCreatePolicyTest(policyPath, common.GvrOperatorPolicy)
+
+			By("Create policy " + policyName1 + " which contains " + operatorPolicyName1)
+			common.DoCreatePolicyTest(policyPath1, common.GvrOperatorPolicy)
+		})
+
+		AfterAll(func() {
+			_, err := common.OcHub(
+				"delete", "-f", policyPath,
+				"-n", userNamespace, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = common.OcHub(
+				"delete", "-f", policyPath1,
+				"-n", userNamespace, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = common.OcManaged(
+				"delete", "ns", subscriptionNamespace, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = common.OcManaged(
+				"delete", "ns", subscriptionNamespace1, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		// RHACM4K-48381 - Test the policy status is Noncompliant when the operator has an upgrade available after
+		// set upgradesAvailable as Noncompliant in the policy
+
+		It(operatorPolicyName+" should be Compliant and OperatorGroup and Subscription should be created", func() {
+			By("Checking if the status of the operator policy is Compliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(Equal(policiesv1.Compliant))
+			By("Checking the status messages of the operator policy")
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(SatisfyAll(
+				ContainSubstring("the OperatorGroup matches what is required by the policy"),
+				ContainSubstring("the Subscription matches what is required by the policy"),
+				ContainSubstring("all operator Deployments have their minimum availability")))
+		})
+
+		It(operatorPolicyName+" should have installed the intended operator "+operatorVersionInitial, func() {
+			By("Checking status of the operator policy")
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("ClusterServiceVersion (" + operatorVersionInitial +
+				") - install strategy completed with no errors"))
+		})
+
+		It(operatorPolicyName+" should be modified to report NonCompliance when upgrades are available", func() {
+			// new policy specifies NonCompliant for upgradesAvailable in complianceConfig
+			By("Patching upgradesAvailable complianceConfig on the operator policy")
+			_, err := common.OcHub("apply", "-f", policyPathModified, "-n", userNamespace)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should become NonCompliant after modifications to complianceConfig", func() {
```

Comment:
This `It` should be combined with the one above it since they are part of the same test.